File: 백준문제풀이/Python/Stack/10828.py
# 명령의 수
import sys
N = int(sys.stdin.readline())
stack = []

for i in range(N):
    command = sys.stdin.readline().split()
# push X: 정수 X를 스택에 넣는 연산이다.
# pop: 스택에서 가장 위에 있는 정수를 빼고, 그 수를 출력한다. 만약 스택에 들어있는 정수가 없는 경우에는 -1을 출력한다.
# size: 스택에 들어있는 정수의 개수를 출력한다.
# empty: 스택이 비어있으면 1, 아니면 0을 출력한다.
# top: 스택의 가장 위에 있는 정수를 출력한다. 만약 스택에 들어있는 정수가 없는 경우에는 -1을 출력한다.
# push처리
    if command[0] == 'push':
        stack.append(command[1])
# pop처리
    elif command[0] == 'pop':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())
# size처리
    elif command[0] == 'size':
        print(len(stack))
# empty처리
    elif command[0] == 'empty':
        if(len(stack)) == 0:
            print(1)
        else:
            print(0)
# top처리
    elif command[0] == 'top':
        if(len(stack)) == 0:
            print(-1)
        else:
            print(stack[-1])

File: 백준문제풀이/C/스택/10828.c
#include <stdio.h>
#include <string.h>

#define MAX_SIZE 10000
#define TRUE 1
#define FALSE 0

int stack[MAX_SIZE];
int idx = -1;

int main(void)
{
    int n, i;

    scanf("%d", &n);
    for (i = 0; i < n; i++)
    {
        char command[6];
        scanf("%s", command);

        if (strcmp(command, "push") == 0)
        {
            int x;
            scanf("%d\n", &x);
            push(x);
        }
        else if (strcmp(command, "pop") == 0)
        {
            printf("%d\n", pop());
        }
        else if (strcmp(command, "size"))
        {
            printf("%d\n", size());
        }
        else if (strcmp(command, "empty"))
        {
            printf("%d\n", empty());
        }
        else if (strcmp(command, "top") == 0)
        {
            printf("%d\n", top());
        }
    }
    return 0;
}

int push(int x)
{
    stack[idx + 1] = x;
    idx++;
}

int pop()
{
    if (empty())
        return -1;
    else
    {
        idx--;
        return stack[idx - 1];
    }
}

int size()
{
    return idx + 1;
}

int empty()
{
    if (idx == -1)
    {
        return TRUE;
    }
    else
        return FALSE;
}

int top()
{
    if (idx == -1)
        return -1;
    else
        return stack[idx];
}